### ISSUE TITLE
Unable to modify labels during createOrReplace

### ISSUE THREAD

**Issue by traviswinter** [fabric8io/kubernetes-client#1468]
I noticed today that we I do a call to createOrUpdate on a resource it will not update if the change is in the metadata. I tracked the issue to ResourceCompare where in trim() the STATUS and METADATA is removed. The status makes sense, but the metadata does not. I understand some distros will add annotations, etc. Was this a quick way to avoid that?

kubeClient.resource(resourceObj).createOrReplace();

I think that adding a new label is a valid scenario for pretty much any resource - does anyone know a way to do it?  Is there any specific reason we remove metadata entirely?

**Comment by brokenjpl** [comment]
The code in question in ResourceCompare:


```
public static <T>  boolean equals(T left, T right) {
        HashMap<String, Object> leftMap = trim((Map<String, Object>) JSON_MAPPER.convertValue(left, TYPE_REF));
        HashMap<String, Object> rightMap = trim((Map<String, Object>) JSON_MAPPER.convertValue(right, TYPE_REF));
        return leftMap.equals(rightMap);
    }

private static HashMap<String, Object> trim(Map<String, Object> map) {
        HashMap<String, Object> result = new HashMap<>(map);
        result.remove(STATUS);
        result.remove(METADATA);
        return result;
    }

```
We could change
` return leftMap.equals(rightMap);`
to (pseudo-code)
` return leftMap.equals(rightMap) && left.labels == right.labels`;

**Comment by rohanKanojia** [comment]
@brokenjpl : Hi, Thanks for reporting. Since you've already found the issue. Could you please raise a PR to fix this issue?

**Comment by brokenjpl** [comment]
Sounds like you're happy with the proposed solution. I'll submit a PR shortly.

**Comment by rohanKanojia** [comment]
closed via #1487
